contributing.md

```diff
@@ -0,0 +1,59 @@
+	 -  This helps to stop you from wasting your time duplicating the work of others that maybe working on the same issue; at the same time
+	 - This step also allows you to get helpful pointers on the community on some problems they may have encountered on similar issues
+ 3. **Fork** the repository
+	 - A fork creates a copy of the code on your Github, so you can work on it separately from everyone else.
+	 - You can learn more about forking [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+ 4. Ensure that you have **commit signing** enabled
+	 - This ensures that the code you submit was committed by you and not someone else who claims to be you
+	 - You can learn more about how to setup commit signing [here](https://www.freecodecamp.org/news/what-is-commit-signing-in-git/ "https://www.freecodecamp.org/news/what-is-commit-signing-in-git/")
+	 - If you have already made some commits that you wish to put in a pull request without signing them, then you can follow [this guide](https://dev.to/jmarhee/signing-existing-commits-with-gpg-5b58) on how to fix that.
+ 5. **Clone** the repository to your your computer.
+	 - This puts a copy of your fork on your computer so you can edit it
+	 - You can learn more about cloning repositories [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+ 6. **Build** the project
+	 - For a detailed look on how to build {{project-name}} look at our [README file](<link to readme>)
+ 7. **Start writing** the code you wanted
+	 - Open up your favorite code editor and make the changes that you wanted to make to the repository
+	 - Make sure to test your code with `{{test-code-cmd}}`
+ 8. **Write tests** for your code
+	 - If you are adding a new feature you should write tests that ensure that if someone make changes to the code it cannot break your new feature without breaking the test
+	 - If your code adds a new feature you should also write at least one documentation test. The documentation test's purpose is to demonstrate and document how to use the API feature
+	 - If your code fixes a bug you should write tests that ensure that if someone makes code changes in the future the bug does not re-emerge without breaking tests
+	 - {% if wasm %}Tests should be put in the test folder and in a test file relevant to the project you are working in {% end if %}
+	 - {% if !wasm %}Tests should be put inside the test folder inside the source folder and in a test file relevant to the project you are working in {% end if %}
+	 - For more information on how to write tests look at [this link](https://doc.rust-lang.org/book/ch11-01-writing-tests.html)
+ 9. Ensure that the code that you made follows our rust **coding guide lines**
+	 - You can find our list of guidelines [here](<link to coding guidelines>)
+	 - This is a courtesy to the programmers that come after you. The easier your code is to read, the easier it will be for the next person to make modifications.
```

Comment:
no `-`? Does it continue from before?

Reply:
I am sorry, I am not sure what you mean here. If you mean  `The easier your code is to read, the easier it will be for the next person to make modifications`. Then it definitely seems like a single Item to me

Reply:
@BoisterousCoder sorry for the confusion. Upon re-read, I meant something more like this:

```markdown
9. Ensure that the code that you made follows our rust **coding guide lines**. 

   - You can find our list of guidelines [here](<link to coding guidelines>). This is 
     a courtesy to the programmers that come after you. The easier your code is to read, 
     the easier it will be for the next person to make modifications.
```

Mainly, it seems this can read as one note for #9 because it's all referencing the link?